api/cluster.go

```diff
@@ -382,12 +388,12 @@ func (s *Server) peerQuerySpeculative(ctx context.Context, data cluster.Traceabl
 			delete(pendingResponses, resp.shardGroup)
 			delete(originalPeers, resp.data.peer.GetName())
 
-		case <-specCheckTicker.C:
+		case <-tickChan:
 			// Check if it's time to speculate!
-			percentReceived := 1 - (float64(len(pendingResponses)) / float64(len(peerGroups)))
-			if percentReceived > speculationThreshold {
+			percentReceived := float64(len(receivedResponses)) / float64(len(peerGroups))
```

Comment:
that's changing the logic, but it's actually making it correct, so that's good.

Reply:
What's the change in logic?

Reply:
this particular line should not change logic. but the line below it does.
i explained in depth in the commit message
(i generally try to make commit messages explain everything)

Reply:
Yes, that was my analysis as well 👍